vaccine_feed_ingest/runners/tx/arcgis/normalize.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python
+import datetime
+import json
+import logging
+import os
+import pathlib
+import re
+import sys
+from typing import List, Optional
+
+from pydantic import ValidationError
+
+# import schema
+site_dir = pathlib.Path(__file__).parent
+state_dir = site_dir.parent
+runner_dir = state_dir.parent
+root_dir = runner_dir.parent
+sys.path.append(str(root_dir))
+
+from schema import schema  # noqa: E402
+
+
+def _get_id(site: dict) -> str:
+    data_id = site["attributes"]["OBJECTID"]
+
+    # Could parse these from directory traversal, but do not for now to avoid
+    # accidental mutation.
+    site = "arcgis"
+    runner = "tx"
+
+    # Could parse these from the input file name, but do not for now to avoid
+    # accidental mutation.
+    arcgis = "3078b524189848569f62985d71f4584b"
+    layer = 0
+
+    return f"{runner}:{site}:{arcgis}_{layer}:{data_id}"
+
+
+def _get_fallback_street(address: str) -> Optional[str]:
+    """
+    Extract the street from the address by taking everything up to the first comma
+    """
+    if not address:
+        return None
+    match = re.search(r"(.*?),", address, flags=re.IGNORECASE)
+    return match.group(1) if match else address
+
+
+def _get_fallback_zip(address: str) -> Optional[str]:
+    """
+    Extract the ZIP from the address if the last 5 characters are digits
+    """
+    if not address:
+        return None
+    match = re.search(r"([0-9]{5})$", address, flags=re.IGNORECASE)
+    return match.group(1) if match else None
+
+
+def _get_fallback_city(address) -> Optional[str]:
+    """
+    Attempt to extract the city from the address. First use the word immediately preceding ", TX" if present. As a
+    secondary fallback, return the word immediately preceding a 5 digit zipcode
+    """
+    if not address:
+        return None
+    match = re.search(r"([\S]*),\s?TX", address, flags=re.IGNORECASE)
+    if match:
+        return match.group(1)
+    match = re.search(r"([\S]*)\s[0-9]{5}", address, flags=re.IGNORECASE)
+    return match.group(1) if match else None
+
+
+def _get_address(site_attributes: dict) -> Optional[schema.Address]:
```

Comment:
Thank you for parsing the addresses and adding clear logs when something does not parse!